File: Python_Practice/LeetCode/0026 Remove Duplicates from Sorted Array/0026-Remove-Duplicates-from-Sorted-Array.py
class Solution(object):
    def removeDuplicates(self, nums):
        x = 0
        for i in range(1, len(nums)):
            if nums[i] != nums[i-1]:
                nums[x] = nums[i-1]
                x+=1
        nums[x] = nums[len(nums)-1]
        x+=1
        return x

        

File: C++_Practice/LeetCode/0026 Remove Duplicates from Sorted Array/0026-Remove-Duplicates-from-Sorted-Array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        unordered_map<int, int> seen;
        int k = 0;
        for (int i = 0; i < nums.size(); i++){
            if(!seen.contains(nums[i])){
                nums[k] = nums[i];
                k++;
            }
            seen[nums[i]] = i;
        }
        return k;
    }
};